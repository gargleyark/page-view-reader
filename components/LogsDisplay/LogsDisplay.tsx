import React from "react";
import useLogs from '../../context/LogsContext/useLogs';
import AllPageViews from '../AllPageViews';
import UniquePageViews from '../UniquePageViews';
import styles from '../../styles/LogsDisplay.module.css';

const LogsDisplay = (): JSX.Element =>  {
  const { logs } = useLogs();

  return (
    <div className={styles.pageDataWrapper}>
      <div className={styles.pageDataColumns}>
      {
        logs.length ? 
          (<>
            <AllPageViews />
            <UniquePageViews />
          </>) :
          'Please upload your logs to see data breakdown'
      }
      </div>
    </div>
  );
}

export default LogsDisplay;