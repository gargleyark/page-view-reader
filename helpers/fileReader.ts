import { Log } from '../context/LogsContext/LogsContext';

export const formatLogs = (logs: string): Array<Log> => {
    return logs
        .split(/\n/)
        .filter(exists => !!exists)
        .map(log => ({
            page: log.split(' ')[0],
            ip: log.split(' ')[1],
        }))
}

export const getFileContent = (file: Blob) => {
    return new Promise<Array<Log>>((resolve) => {
        var fileReader = new FileReader();  
        fileReader.onloadend = () => resolve(formatLogs(`${fileReader.result}`));
        fileReader.onerror = () => resolve([]);
        fileReader.readAsText(file);
    });
};