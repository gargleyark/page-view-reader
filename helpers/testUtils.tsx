import { render } from "@testing-library/react";
import LogsContext, { Log } from '../context/LogsContext/LogsContext'

const Providers = ({ children }: { children: JSX.Element}) => {
  const logs: Array<Log> = [
    {
      page: '/home',
      ip: '1'
    },
    {
      page: '/home',
      ip: '1'
    },
    {
      page: '/about',
      ip: '1'
    },
    {
      page: '/about',
      ip: '1'
    },
    {
      page: '/about',
      ip: '2'
    },
    {
      page: '/about',
      ip: '2'
    }
  ];

  const setLogs = () => {};

  return (
    <LogsContext.Provider value={{ logs, setLogs }}>
      { children }
    </LogsContext.Provider >
  );
};

const customRender = (ui: JSX.Element, options: Record<string, string> = {}) =>
  render(ui, { wrapper: Providers, ...options });

export * from "@testing-library/react";

export { customRender as render };