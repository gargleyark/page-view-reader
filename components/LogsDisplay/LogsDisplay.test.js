import React from "react";
import { render, screen } from "../../helpers/testUtils";
import LogsDisplay from ".";

describe("Logs display", () => {
    it('should render correctly', () => {
        const { container } = render(<LogsDisplay />);
        expect(container).toMatchSnapshot();
    });
});