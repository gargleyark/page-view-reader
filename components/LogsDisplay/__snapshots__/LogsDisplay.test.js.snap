// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logs display should render correctly 1`] = `
<div>
  <div>
    <div>
      <div>
        <h2>
          All page views
        </h2>
        <div>
          <span>
            /about
          </span>
          :
           
          <span>
            4
             
            page views
          </span>
        </div>
        <div>
          <span>
            /home
          </span>
          :
           
          <span>
            2
             
            page views
          </span>
        </div>
      </div>
      <div>
        <h2>
          Unique page views
        </h2>
        <div>
          <span>
            /about
          </span>
          :
           
          <span>
            2
             
            unique views
          </span>
        </div>
        <div>
          <span>
            /home
          </span>
          :
           
          <span>
            1
             
            unique views
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
